src/ui/components/SubmitProposal.tsx

```diff
@@ -29,7 +32,13 @@ export function SubmitProposal() {
     dispatch({type: 'RESET'});
   };
 
-  const isDisabled = !state.account || !state.preimageHash || !state.balance;
+  const enteredBalance = stringToBn(state.balance ?? '0') ?? BN_ZERO;
```

Comment:
Why do you need to add `?? BN_ZERO` ? `stringToBn` returns `BN`

Reply:
in this case `state.balance` return string | undefined despite it being set to string. since I'm not comparing the value of undefined so I used `??` to handle the undefined cases.

Reply:
I understand that `state.balance` can be `undefined`, but the result of` stringToBn(state.balance ?? '0') ` would not be `undefined`, so `?? BN_ZERO` should not be needed.

Reply:
The result from `stringToBn` returns `BN` or `undefined`, thereby I had to do it like that. in another case, `formattedStringToBn` returns only `BN` I fixed for that one.

Reply:
How can `stringToBn` return `undefined`? Here is the definition: 

`export function stringToBn(registry: Registry, input: string): BN `

Reply:
In this case, I'm using `stringToBn` from the `useFormatBalance` hook and not from Utils, but under the hood, it uses the `StringToBn` from utils.

Reply:
Got it